field_manager/field_manager.F90

```diff
@@ -560,6 +555,39 @@ subroutine field_manager_init(nfields, table_name)
 integer,                      intent(out), optional :: nfields    !< number of fields
 character(len=fm_string_len), intent(in),  optional :: table_name !< Name of the field table, default
 
+if (module_is_initialized) then
+   if(present(nfields)) nfields = num_fields
+   return
+endif
+
+call initialize
+
+if (use_field_table_yaml) then
```

Comment:
You have _!TODO_ comments throughout the code, but not one here to flag this for area for eventual removal.